### ISSUE TITLE
Failed to create azurerm_iothub_route with source DeviceConnectionStateEvents in v2.72.0 

### ISSUE THREAD

**Issue by agriessbosch** [hashicorp/terraform-provider-azurerm#12985]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

> Terraform v1.0.4
> on linux_amd64
> + provider registry.terraform.io/hashicorp/azurerm v2.72.0

### Affected Resource(s)

* `azurerm_iothub_route`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "devconnstatefail" {
  name     = "devconnstatefail-resources"
  location = "West Europe"
}

resource "azurerm_eventhub_namespace" "devconnstatefail" {
  name                = "devconnstatefail-namespace"
  resource_group_name = azurerm_resource_group.devconnstatefail.name
  location            = azurerm_resource_group.devconnstatefail.location
  sku                 = "Basic"
}

resource "azurerm_eventhub" "devconnstatefail" {
  name                = "devconnstatefail-eventhub"
  resource_group_name = azurerm_resource_group.devconnstatefail.name
  namespace_name      = azurerm_eventhub_namespace.devconnstatefail.name
  partition_count     = 2
  message_retention   = 1
}

resource "azurerm_eventhub_authorization_rule" "devconnstatefail" {
  resource_group_name = azurerm_resource_group.devconnstatefail.name
  namespace_name      = azurerm_eventhub_namespace.devconnstatefail.name
  eventhub_name       = azurerm_eventhub.devconnstatefail.name
  name                = "acctest"
  send                = true
}

resource "azurerm_iothub" "devconnstatefail" {
  name                = "devconnstatefail-IoTHub"
  resource_group_name = azurerm_resource_group.devconnstatefail.name
  location            = azurerm_resource_group.devconnstatefail.location

  sku {
    name     = "S1"
    capacity = "1"
  }
}

resource "azurerm_iothub_endpoint_eventhub" "endpoint_eventhub" {
  resource_group_name = azurerm_resource_group.devconnstatefail.name
  iothub_name         = azurerm_iothub.devconnstatefail.name
  name                = "devconnstatefail"

  connection_string = azurerm_eventhub_authorization_rule.devconnstatefail.primary_connection_string
}

resource "azurerm_iothub_route" "eventhub_route" {
  resource_group_name = azurerm_resource_group.devconnstatefail.name
  iothub_name         = azurerm_iothub.devconnstatefail.name
  name                = "devconnstatefail"

  source         = "DeviceConnectionStateEvents"
  condition      = "true"
  endpoint_names = [azurerm_iothub_endpoint_eventhub.endpoint_eventhub.name]
  enabled        = true
}
```

### Expected Behaviour

A route with the source DeviceConnectionStateEvents should be created.

### Actual Behaviour

The following error is produced:

> azurerm_iothub_route.eventhub_route: Creating...
╷
│ Error: creating/updating IotHub "devconnstatefail-IoTHub" (Resource Group "devconnstatefail-resources"): devices.IotHubResourceClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="Failed" Message="The async operation failed." InnerError={"unmarshalError":"json: cannot unmarshal number into Go struct field serviceErrorInternal.code of type string"} AdditionalInfo=[{"code":400024,"httpStatusCode":"BadRequest","message":"DeviceConnectionStateEvents Routing Source routing properties are only supported in api version 2021-02-01-preview or above. If you contact a support representative please include this correlation identifier: 8b1d81b3-4842-4f9e-88b9-f57f24b19a99, timestamp: 2021-08-13 08:57:17Z, errorcode: IH400024."}]
│ 
│   with azurerm_iothub_route.eventhub_route,
│   on main.tf line 63, in resource "azurerm_iothub_route" "eventhub_route":
│   63: resource "azurerm_iothub_route" "eventhub_route" {
│ 
╵

### Steps to Reproduce

1. `terraform apply`

### References

* #12477

**Comment by ramondeklein** [comment]
I have created a PR to upgrade to the new API (#13324).

**Comment by catriona-m** [comment]
Thanks for raising this @agriessbosch. This issue can be closed as it has since been fixed in #13324.
